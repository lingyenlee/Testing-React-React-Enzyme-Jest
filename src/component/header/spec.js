import React from "react";
import { shallow } from "enzyme";
import Header from "./index";
import { findByTestAttr } from "../../../Utils";

const setUp = (props = {}) => {
  const component = shallow(<Header {...props} />);
  return component;
};

describe("Header Component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should render without error", () => {
    const wrapper = findByTestAttr(component, "HeaderComponent");
    expect(wrapper.length).toBe(1);
  });
  it("should render a logo", () => {
    const wrapper = findByTestAttr(component, "logoIMG");
    expect(wrapper.length).toBe(1);
  });
});
