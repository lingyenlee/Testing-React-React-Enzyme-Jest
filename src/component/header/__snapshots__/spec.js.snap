// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot 1`] = `
<header
  data-test="HeaderComponent"
>
  <div
    className="wrap"
  >
    <div
      className="logo"
    >
      <img
        alt="Logo"
        data-test="logoIMG"
        src="logo1.png"
      />
    </div>
  </div>
</header>
`;
